pubsub/api/PubsubSample/Program.cs

```diff
@@ -236,28 +237,26 @@ public static object CreateTopic(string projectId, string topicId)
             return 0;
         }
 
-        public static PublisherClient GetPublisher(string projectId,
+        public static async Task<PublisherClient> GetPublisherAsync(string projectId,
             string topicId)
         {
             // [START pubsub_publish]
-            PublisherServiceApiClient publisherClient = PublisherServiceApiClient.Create();
-            PublisherClient publisher = PublisherClient.Create(
-                new TopicName(projectId, topicId), new[] { publisherClient });
+            PublisherClient publisher = await PublisherClient.CreateAsync(

```

Comment:
What's the point to await creation of the client and then return it right away? You don't really need the async context to continue the execution of the method (typical reason for using await)
Why not just making the method asynchronous without async..await?

```csharp

public static Task<PublisherClient> GetPublisherAsync(
            string projectId,
            string topicId)
{
   // [START pubsub_publish]
   Task<PublisherClient> publisherTask = PublisherClient.CreateAsync(
         new TopicName(projectId, topicId));
  // [END pubsub_publish]
  return publisherTask;
}
```

Reply:
When the code appears in the documentation, it won't have the full context of the function call.  In that context, I think it's more useful to show the user a call that awaits the PublisherClient.